src/node.cc

```diff
@@ -917,68 +789,99 @@ void ReportException(Environment* env,
   if (message.IsEmpty())
     message = Exception::CreateMessage(env->isolate(), er);
 
-  AppendExceptionLine(env, er, message, FATAL_ERROR);
-
-  Local<Value> trace_value;
-  Local<Value> arrow;
-  const bool decorated = IsExceptionDecorated(env, er);
-
-  if (er->IsUndefined() || er->IsNull()) {
-    trace_value = Undefined(env->isolate());
+  if (!er->IsObject()) {
+    ScriptOrigin origin = message->GetScriptOrigin();
+    node::Utf8Value filename(env->isolate(), message->GetScriptResourceName());
+    const char* filename_string = *filename;
+    int linenum = message->GetLineNumber(env->context()).FromJust();
+    // Print line of source code.
+    MaybeLocal<String> source_line_maybe =
+      message->GetSourceLine(env->context());
+    node::Utf8Value sourceline(env->isolate(),
+                               source_line_maybe.ToLocalChecked());
+    const char* sourceline_string = *sourceline;
+    int script_start =
+      (linenum - origin.ResourceLineOffset()->Value()) == 1 ?
+          origin.ResourceColumnOffset()->Value() : 0;
+    int start = message->GetStartColumn(env->context()).FromMaybe(0);
+    int end = message->GetEndColumn(env->context()).FromMaybe(0);
+    if (start >= script_start) {
+      CHECK_GE(end, start);
+      start -= script_start;
+      end -= script_start;
+    }
+    char out[1024];
```

Comment:
IMHO this would simpler with `std::stringstream`. No need to keep track of sizes and offset, no danger of buffer overruns.